app/variablesmanager.cpp

```diff
@@ -0,0 +1,41 @@
+#include "variablesmanager.h"
+
+#include "qgsexpressioncontextutils.h"
+
+VariablesManager::VariablesManager( MerginApi *merginApi, QObject *parent )
+  : QObject( parent )
+  , mMerginApi( merginApi )
+{
+  QgsExpressionContextUtils::setGlobalVariable( QStringLiteral( "mergin_url" ),  mMerginApi->apiRoot() );
+
+  QObject::connect( mMerginApi, &MerginApi::apiRootChanged, this, &VariablesManager::apiRootChanged );
+  QObject::connect( merginApi, &MerginApi::authChanged, this, &VariablesManager::authChanged );
+}
+
+void VariablesManager::apiRootChanged()
+{
+  QgsExpressionContextUtils::setGlobalVariable( QStringLiteral( "mergin_url" ),  mMerginApi->apiRoot() );
+}
+
+void VariablesManager::authChanged()
+{
+  QgsExpressionContextUtils::setGlobalVariable( QStringLiteral( "mergin_username" ),  mMerginApi->merginUserName() );
+}
+
+void VariablesManager::merginProjectChanged( QgsProject *project )
+{
+  QString filePath = project->fileName();
+  LocalProjectInfo info = mMerginApi->localProjectsManager().projectByProjectFilePath( filePath );
```

Comment:
This can return invalid project info if it is not mergin project... I guess we want to return early - and probably also clear project variables if there were any left from the previous project?